novawallet/Modules/Vote/Governance/Validating/GovernanceValidatorFactory.swift

```diff
@@ -166,31 +97,38 @@ extension GovernanceValidatorFactory: GovernanceValidatorFactoryProtocol {
             }
 
             let amountFormatter = self?.assetBalanceFormatterFactory.createTokenFormatter(for: assetInfo)
-            let amountString: String
-            let freeInPlank = assetBalance?.freeInPlank ?? 0
-
-            if
-                let amountDecimal = Decimal.fromSubstrateAmount(freeInPlank, precision: assetInfo.assetPrecision) {
-                amountString = amountFormatter?.value(for: locale ?? Locale.current).stringFromDecimal(
-                    amountDecimal
-                ) ?? ""
+            let availableForOpenGov = assetBalance?.availableForOpenGov ?? 0
+
+            let amountDecimal = availableForOpenGov.decimal(assetInfo: assetInfo)
+            let amountString = amountFormatter?.value(for: locale ?? Locale.current).stringFromDecimal(
+                amountDecimal
+            ) ?? ""
+
+            if let maxAmountErrorClosure, availableForOpenGov > 0 {
+                self?.presentable.presentNotEnoughTokensToVote(
+                    from: view,
+                    available: amountString,
+                    maxAction: {
+                        maxAmountErrorClosure(availableForOpenGov)
+                    },
+                    locale: locale
+                )
             } else {
-                amountString = ""
+                self?.presentable.presentNotEnoughTokensToVote(
+                    from: view,
+                    available: amountString,
+                    maxAction: nil,
+                    locale: locale
+                )
             }
-
-            self?.presentable.presentNotEnoughTokensToVote(
-                from: view,
-                available: amountString,
-                locale: locale
-            )
         }, preservesCondition: {
             guard
                 let assetBalance = assetBalance,
                 let votingAmount = votingAmount else {
                 return false
             }
 
-            return assetBalance.freeInPlank >= votingAmount
+            return assetBalance.availableForOpenGov >= votingAmount
```

Comment:
`availableForOpenGov` validating `votingAmount` for common gov validation needs looks a bit misleading. Possibly, we should use `freeInPlank` here.

Reply:
Why it is misleading? We are using ```availableForOpenGov``` abstraction to check voting amount in the ```enoughTokenForVoting``` method. I think I have moved to that abstraction to most places in swipegov and it is still uses ```freeInPlank``` under the hood. We have another task to complete migration in all places in the opengov and change the logic behind ```availableForOpenGov```